File: Reflow/Tasks/TaskBase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Serialization;

namespace Reflow.Tasks
{
    public abstract class TaskBase : ITask
    {
        public virtual string Name { get; set; }
        
        public virtual TaskResult Execute(TaskExecutionContext context)
        {
            if (this.OnBeforeExecution != null)
            {
                this.OnBeforeExecution(context, null);
            }
            TaskResult result = this.OnExecute ( context);
            if (this.OnAfterExecution != null)
            {
                this.OnAfterExecution(context, result);
            }
            context.LastTaskResult = result;
            return result;
        }

        public abstract TaskResult OnExecute(TaskExecutionContext context);

        [XmlIgnore]
        public virtual TaskExecutionEventHandler OnBeforeExecution
        {
            get;
            set;
        }

        [XmlIgnore]
        public virtual TaskExecutionEventHandler OnAfterExecution
        {
            get;
            set;
        }
    }
}

File: Java/src/main/java/com/nuarca/etl/tasks/TaskBase.java
//
// Translated by CS2J (http://www.cs2j.com): 12/6/2017 2:34:22 PM
//

package com.nuarca.etl.tasks;

import com.nuarca.etl.tasks.ITask;
import com.nuarca.etl.tasks.TaskExecutionContext;
import com.nuarca.etl.tasks.TaskExecutionEventHandler;
import com.nuarca.etl.tasks.TaskResult;

public abstract class TaskBase   implements ITask
{
    private String __Name;
    public String getName() {
        return __Name;
    }

    public void setName(String value) {
        __Name = value;
    }

    public TaskResult execute(TaskExecutionContext context) throws Exception {
        if (this.getOnBeforeExecution() != null)
        {
            this.getOnBeforeExecution().invoke(context,null);
        }
         
        TaskResult result = this.onExecute(context);
        if (this.getOnAfterExecution() != null)
        {
            this.getOnAfterExecution().invoke(context,result);
        }
         
        context.setLastTaskResult(result);
        return result;
    }

    public abstract TaskResult onExecute(TaskExecutionContext context) throws Exception ;

    private TaskExecutionEventHandler __OnBeforeExecution;
    public TaskExecutionEventHandler getOnBeforeExecution() {
        return __OnBeforeExecution;
    }

    public void setOnBeforeExecution(TaskExecutionEventHandler value) {
        __OnBeforeExecution = value;
    }

    private TaskExecutionEventHandler __OnAfterExecution;
    public TaskExecutionEventHandler getOnAfterExecution() {
        return __OnAfterExecution;
    }

    public void setOnAfterExecution(TaskExecutionEventHandler value) {
        __OnAfterExecution = value;
    }

}


